src/main/kotlin/org/rust/ide/ssr/RsStructuralSearchProfile.kt

```diff
@@ -34,12 +54,69 @@ class RsStructuralSearchProfile : StructuralSearchProfile() {
 
     override fun createMatchingVisitor(globalVisitor: GlobalMatchingVisitor): PsiElementVisitor = RsMatchingVisitor(globalVisitor)
 
-    override fun getPredefinedTemplates(): Array<Configuration> = RsPredefinedConfigurations.createPredefinedTemplates()
+    override fun getPredefinedTemplates(): Array<Configuration> {
+        if (!isFeatureEnabled(RsExperiments.SSR)) return arrayOf()
+        return RsPredefinedConfigurations.createPredefinedTemplates()
+    }
 
     override fun isIdentifier(element: PsiElement?): Boolean = element?.node?.elementType == IDENTIFIER
 
     override fun createCompiledPattern(): CompiledPattern = RsCompiledPattern()
 
+    override fun createPatternTree(
+        text: String,
+        context: PatternTreeContext,
+        fileType: LanguageFileType,
+        language: Language,
+        contextId: String?,
+        project: Project,
+        physical: Boolean
+    ): Array<PsiElement> {
+        var patternTree = super.createPatternTree(text, context, fileType, language, contextId, project, physical)
+        if (patternTree.firstOrNull() is PsiErrorElement
+            || patternTree.size > 1 && patternTree[0] is LeafPsiElement && patternTree[1] is PsiErrorElement
+            || patternTree.firstOrNull() is DummyBlock) {
+            val factory = RsPsiFactory(project)
+            patternTree = factory.createBlockExpr(text).firstChild.children
+        }
+//        for (element in patternTree) print(DebugUtil.psiToString(element, false))
```

Comment:
```suggestion
```

Reply:
Can we leave it? This line is really really useful as a debug tool, it prints the template in the console for tests or in IDE

Reply:
Ok let's keep